### PR TITLE
add support for user.js to beez and protostar

### PR DESCRIPTION
**Pull request by brianteeman**
Pull Request for Issue #12683 .

### Summary of Changes
We have support for user.css - this adds support or an equivalent js file

### Testing Instructions
Create a user.js file with some content in /templates/beez3/js
Create a user.js file with some content in /templates/protostar/js
and check that they are loaded

Remove the user.js files and check that the template is not trying to load them

### CODE REVIEW

File: templates/beez3/index.php [hunk 1]
```diff
@@ -85,6 +85,14 @@
 $this->addScript($this->baseurl . '/templates/' . $this->template . '/javascript/respond.src.js');
 $this->addScript($this->baseurl . '/templates/' . $this->template . '/javascript/template.js');
 
+// Check for a custom js file
+$userJs = JPATH_SITE . '/templates/' . $this->template . '/js/user.js';
+
+if (file_exists($userJs && filesize($userJs) > 0)
+{
+	$this->addScriptVersion($this->baseurl . '/templates/' . $this->template . '/js/user.js');
```
Comment (andrepereiradasilva) [inline templates/beez3/index.php]: please use (in all)
```php
JHtml::_('script', 'user.js', array('version' => 'auto', 'relative' => true));
```
So i don't have to modify this too after https://github.com/joomla/joomla-cms/pull/12373

Comment (brianteeman) [inline templates/beez3/index.php]: so just to be clear. 
replace line 93 with that line?

File: templates/beez3/index.php [hunk 2]
```diff
@@ -85,6 +85,14 @@
 $this->addScript($this->baseurl . '/templates/' . $this->template . '/javascript/respond.src.js');
 $this->addScript($this->baseurl . '/templates/' . $this->template . '/javascript/template.js');
 
+// Check for a custom js file
+$userJs = JPATH_SITE . '/templates/' . $this->template . '/js/user.js';
+
+if (file_exists($userJs && filesize($userJs) > 0)
```
Comment (ggppdk) [inline templates/beez3/index.php]: This code open 3 parenthesis and closes 2 ?
I think, you meant something like:
```php
if ( JFile::exists($userJs) && filesize($userJs))
```
Same for the other file

**Comment by brianteeman** [comment]
Thanks I will check later. It should just have been a copy paste of the
existing css code

File: templates/beez3/index.php (hunk 1 above)
Comment (andrepereiradasilva) [inline templates/beez3/index.php]: IIRC JHtml script relative checks the existence of the file already (in that case the if is not needed). but do test

File: templates/beez3/index.php (hunk 2 above)
Comment (brianteeman) [inline templates/beez3/index.php]: fixed the missing bracket

File: templates/beez3/index.php (hunk 1 above)
Comment (brianteeman) [inline templates/beez3/index.php]: will look at this change later today or tomorrow

**Comment by brainforgeUK** [comment]
Should this be included with ISIS and other admin templates?
See #12683 referred to above.

**Comment by brianteeman** [comment]
Personally I don't see the need for this in the admin.

**Comment by ghazal** [comment]
I have tested this item :white_check_mark: successfully on d39ed90ea9f5e5fd6b36fbbde5ff9badf2156353<br /><br />Being a heavy JS user, I really appreciate this.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12684">issues.joomla.org/tracker/joomla-cms/12684</a>.</sub>

**Comment by andrepereiradasilva** [comment]
I have tested this item :white_check_mark: successfully on fc0e3820b989d8cd400ece0d974c442074ec450a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12684">issues.joomla.org/tracker/joomla-cms/12684</a>.</sub>

**Comment by brianteeman** [comment]
Updated with version code from @andrepereiradasilva

**Comment by andrepereiradasilva** [comment]
@ghazal you need to retest after latest changes

**Comment by ghazal** [comment]
I have tested this item :white_check_mark: successfully on fc0e3820b989d8cd400ece0d974c442074ec450a<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12684">issues.joomla.org/tracker/joomla-cms/12684</a>.</sub>

**Comment by zero-24** [comment]
RTC cc: @rdeutz <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/12684">issues.joomla.org/tracker/joomla-cms/12684</a>.</sub>

**Comment by rdeutz** [comment]
I might be wrong but the usual convention for a javascript directory is "js"

**Comment by brianteeman** [comment]
It is but you will see that beez is called /javascript/ and always has been - I guess you forgot

**Comment by rdeutz** [comment]
ah getting old :-)

**Comment by brianteeman** [comment]
Thanks

**Pull request merged**